library/lua/gui.lua

```diff
@@ -920,14 +920,42 @@ local function make_frame(name, double_line)
     end
 
     local frame = copyall(BASE_FRAME)
+    -- external horizontal/vertical bars
     frame.t_frame_pen = to_pen{ tile=tp(1), ch=double_line and 205 or 196, fg=COLOR_GREY, bg=COLOR_BLACK }
     frame.l_frame_pen = to_pen{ tile=tp(7), ch=double_line and 186 or 179, fg=COLOR_GREY, bg=COLOR_BLACK }
     frame.b_frame_pen = to_pen{ tile=tp(15), ch=double_line and 205 or 196, fg=COLOR_GREY, bg=COLOR_BLACK }
     frame.r_frame_pen = to_pen{ tile=tp(9), ch=double_line and 186 or 179, fg=COLOR_GREY, bg=COLOR_BLACK }
+    -- external corners
     frame.lt_frame_pen = to_pen{ tile=tp(0), ch=double_line and 201 or 218, fg=COLOR_GREY, bg=COLOR_BLACK }
     frame.lb_frame_pen = to_pen{ tile=tp(14), ch=double_line and 200 or 192, fg=COLOR_GREY, bg=COLOR_BLACK }
     frame.rt_frame_pen = to_pen{ tile=tp(2), ch=double_line and 187 or 191, fg=COLOR_GREY, bg=COLOR_BLACK }
     frame.rb_frame_pen = to_pen{ tile=tp(16), ch=double_line and 188 or 217, fg=COLOR_GREY, bg=COLOR_BLACK }
+    -- external T-junctions
+    frame.tte_frame_pen = to_pen{ tile=tp(10), ch=double_line and 203 or 194, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.bte_frame_pen = to_pen{ tile=tp(11), ch=double_line and 202 or 193, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.lte_frame_pen = to_pen{ tile=tp(12), ch=double_line and 204 or 195, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.rte_frame_pen = to_pen{ tile=tp(13), ch=double_line and 185 or 180, fg=COLOR_GREY, bg=COLOR_BLACK }
+    -- internal horizontal/vertical bars (and cross junction)
+    frame.v_frame_pen = to_pen{ tile=tp(4), ch=double_line and 186 or 179, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.h_frame_pen = to_pen{ tile=tp(5), ch=double_line and 205 or 196, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.x_frame_pen = to_pen{ tile=tp(3), ch=double_line and 206 or 197, fg=COLOR_GREY, bg=COLOR_BLACK }
+    -- internal T-junctions
+    frame.tti_frame_pen = to_pen{ tile=tp(20), ch=double_line and 203 or 194, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.bti_frame_pen = to_pen{ tile=tp(19), ch=double_line and 202 or 193, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.lti_frame_pen = to_pen{ tile=tp(18), ch=double_line and 204 or 195, fg=COLOR_GREY, bg=COLOR_BLACK }
+    frame.rti_frame_pen = to_pen{ tile=tp(17), ch=double_line and 185 or 180, fg=COLOR_GREY, bg=COLOR_BLACK }
+    --[[ naming scheme for t-junctions:
+        the first letter means:
+            "t" for "top":    the T is not rotated, it looks like the character "T"
+            "b" for "bottom": the T is upside-down
+            "l" for "left":   the T is rotated 90 degrees counter-clockwise
+            "r" for "right":  the T is rotated 90 degrees clockwise
+        the second letter means:
+            "t" for "t-junction": this is a t-junction piece...
+        the third letter means:
+            "i" for "internal": this sprite is meant to be drawn inside the frame.
+            "e" for "external": this sprite is meant to be drawn on the frame itself.
+    --]]
```

Comment:
idk if this is necessary but ive included it for now

Reply:
I certainly needed the explanation. I'd say keep it.

Reply:
In fact, maybe add a little explanation about the difference between an external and internal t-junction (which I only now understand because I looked at the sprite sheet).

Reply:
right. yeah, will do. 
i've taken their obviousness for granted because i've been analyzing/staring at/editing those spritesheets for a long time, but it's such an unintuitive concept 🤣 
In ascii mode there's no difference between internal/external either!